app/api/v1/vaults.py

```
@@ -112,3 +136,153 @@ async def send_unlock_command(
+
+
+@router.post(
+    "/vaults/{vault_id}/unlock/pin",
+    response_model=UnlockWithPINResponse,
+    status_code=status.HTTP_200_OK,
+)
+async def unlock_vault_with_pin(
+    vault_id: str,
+    payload: UnlockWithPINRequest,
+    current_user_id: str = Depends(get_current_user_id),
+    check_access: CheckVaultAccess = Depends(get_check_vault_access_uc),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)